homeassistant/helpers/check_config.py

```diff
@@ -128,16 +144,16 @@ def _comp_error(ex: Exception, domain: str, config: ConfigType) -> None:
             integration = await async_get_integration_with_requirements(hass, domain)
         except loader.IntegrationNotFound as ex:
             if not hass.config.recovery_mode and not hass.config.safe_mode:
-                result.add_error(f"Integration error: {domain} - {ex}")
+                result.add_warning(f"Integration error: {domain} - {ex}")
```

Comment:
We should maybe still report this in recovery and safe mode, but with a different error message?

Reply:
Can we identify, if the integration is part of the core?
Maybe we could use the `generated/integrations.json` for identification

So we could filter errors from integration which are not loaded in the current mode

Reply:
I'd prefer to implement this in a separate PR